internal/controller/deployment_test.go

```diff
@@ -142,7 +142,7 @@ func TestConstructEnvForApp(t *testing.T) {
 				},
 			}
 
-			envs := ConstructEnvForApp(context.Background(), app)
+			envs := ConstructEnvForApp(context.Background(), app, spinv1alpha1.OtelConfig{})
```

Comment:
Maybe set something in the `OtelConfig` to actually test it is being set properly?